### DISCUSSION TITLE
batch combination of jpg's

### DISCUSSION THREAD

**Discussion by alruben** [ImageMagick/ImageMagick#5315] [Help]
Greetings, I am in process of scanning a very large collection of old photos. Many of them have something written on the back. I am using a scanner that does a duplex scan, but it places from front and rear image in to two separate files. An example of these scans is here:

https://tinyurl.com/bd3z5af7

This directory could have upwards of several hundred scans in it as I am moving along the collection. One thing that can always be relied upon is that there will be an even number of files (front and back), and they will be in order when sorted by file name.

What I am looking to accomplish is as follows. I'd like to run a batch process that takes two pictures at a time (1,2 then 3,4 then 5,6 and so on) and combines the two into a "montage" (for lack of a better term). The resulting jpg should be placed in another directory.

Any pointers on this would be helpful!

Edit: ImageMagick-7.1.0-43-Q16-HDRI-x64-dll.exe on Windows 10.

**Comment by snibgo** [comment]
What version of IM, on what platform? Please always state this.

First, I suggest you figure out a simple IM command that does what you want for a pair of files. Something like this:
```
magick 20220711_011.jpg 20220711_012.jpg -append +repage outdir\20220711_011_comb.jpg
```
You might prefer "+append" or something else.

Then, put that in a shell loop that loops through each pair of files. You need to decide: what should happen when you have an even number without the odd number? Or the odd number without the even number? Are there missing gaps?

**Reply by alruben** [reply]
Sorry - post updated. ImageMagick-7.1.0-43-Q16-HDRI-x64-dll.exe on Windows. 

I have been playing with append. As I said, I can always rely on that there will be 2 files sequentially; there might be 2 files, 50 files, or 468 files in a directory; but if sorted by name, they pairs will always be adjacent. I was hoping there would be a simple way to tell magick to just do two at a time rather than writing a script. My scripting days are, well, a long way behind me. I will have to brush the dust off.

**Comment by snibgo** [comment]
> As I said, I can always rely on that there will be 2 files sequentially; ...

I asked because your sample has missing numbers.

Anyhow, here is a simple Windows BAT script. It probably contains more error checking than you need.
```
set SRCDIR=prod
set DSTDIR=outfiles

set PREF=20220711
set EXT=jpg

if not exist %DSTDIR% md %DSTDIR%

setlocal enabledelayedexpansion

rem Don't turn echo off until we know it does what we want.
rem echo off

for /L %%I in (1,2,19) do (
  echo %%I

  set LZ=000%%I
  set LZ=!LZ:~-3!

  set /A LZp1=%%I+1
  set LZp1=000!LZp1!
  set LZp1=!LZp1:~-3!

  echo LZ=!LZ! LZp1=!LZp1!

  if exist %SRCDIR%\%PREF%_!LZ!.%EXT% (

    if exist %SRCDIR%\%PREF%_!LZp1!.%EXT% (
      echo found LZ=!LZ! LZp1=!LZp1!
    ) else (
      echo Can't find even number !LZp1!
    )

    magick %SRCDIR%\%PREF%_!LZ!.%EXT% %SRCDIR%\%PREF%_!LZp1!.%EXT% -append %DSTDIR%\%PREF%_!LZ!_com.%EXT% 

    if ERRORLEVEL 1 (
      echo %0: magick failed
      exit /B 1
    )

  ) else (
    echo Can't find odd number !LZ!
  )
)
```
Ensure you have backups before running this.

The LZ stuff is to make variables with leading zeros.
